File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.Arrays;

class Solution {
    public boolean solution(String[] phone_book) {
        int n = phone_book.length;
        Arrays.sort(phone_book);
        
        for (int i = 0; i < n - 1; i++) {
            if (phone_book[i].startsWith(phone_book[i + 1])) return false;
            if (phone_book[i + 1].startsWith(phone_book[i])) return false;
        }
        
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book = sorted(set(phone_book))
    
    for i in range(len(phone_book) - 1):
        if phone_book[i] == phone_book[i + 1][:len(phone_book[i])]:
            return False
            
    return True

#     phone_book.sort()
    
#     for i in range(len(phone_book)):
#         for j in range(i + 1, len(phone_book)):
#             if phone_book[i][0] != phone_book[j][0]:
#                 pass
            
#             if phone_book[i] == phone_book[j][:len(phone_book[i])]:
#                 return False            
    
#     return True